commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/AddKerberosAuthentication.java

```diff
@@ -0,0 +1,133 @@
+package org.wildfly.extras.creaper.commands.security.realms;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.wildfly.extras.creaper.commands.foundation.offline.xml.GroovyXmlTransform;
+import org.wildfly.extras.creaper.commands.foundation.offline.xml.Subtree;
+import org.wildfly.extras.creaper.core.ManagementVersion;
+import org.wildfly.extras.creaper.core.offline.OfflineCommandContext;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Batch;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+/**
+ * Add kerberos authentication to security realm.
+ *
+ * <b>This command is available since WildFly 9 or in EAP 6.4.x!</b>
+ */
+public final class AddKerberosAuthentication extends AbstractAddSecurityRealmSubElement {
+
+    private final Boolean removeRealm;
+    private final List<KerberosKeytab> keytabs;
+
+    private AddKerberosAuthentication(Builder builder) {
+        super(builder);
+        this.removeRealm = builder.removeRealm;
+        this.keytabs = builder.keytabs;
+    }
+
+    @Override
+    public void apply(OnlineCommandContext ctx) throws Exception {
+        if (ctx.client.serverVersion().lessThan(ManagementVersion.VERSION_3_0_0)
+                && !ctx.client.serverVersion().equalTo(ManagementVersion.VERSION_1_7_0)) {
+            throw new AssertionError("Kerberos authentication in security realm is available since WildFly 9 or in EAP 6.4.x.");
+        }
+
+        Operations ops = new Operations(ctx.client);
+
+        Address securityServerIdentitiesAddress = securityRealmAddress.and("server-identity", "kerberos");
+        Address securityRealmKerberosAuthnAddress = securityRealmAddress.and("authentication", "kerberos");
+
+        if (replaceExisting) {
+            Batch batch = new Batch();
+            boolean krbServerIdentityExists = false;
+            try {
+                krbServerIdentityExists = ops.exists(securityServerIdentitiesAddress);
+            } catch (OperationException e) {
+                throw new IOException("Failed to read kerberos server identity.", e);
```

Comment:
The `apply` method can throw any exception and you even declare `throws Exception`, so why convert `OperationException` to `IOException`?